### PR TITLE
Hil tests pdm pcm

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
### Summary

- Small refactor to get tests working. Currently 16 bit all modes are tested.
- Updated HIL test setup and docs.
- Updated docs and removed non-working bit configs and disabled in code as well.

### CODE REVIEW

File: .github/workflows/ports_psoc6.yml [hunk 1]
```diff
@@ -70,8 +70,10 @@ jobs:
         source tools/ci.sh && ci_psoc6_flash_multiple_devices ${{ matrix.board }} firmware.hex tools/psoc6/${{ runner.name }}-devs.yml
     - name: Run psoc6 tests
       timeout-minutes: 12
-      run: | 
+      run: |
        ./tests/ports/psoc6/run_psoc6_tests.sh --test-suite ci-tests --board ${{ matrix.board }} --hil ${{ runner.name }}
+#./tests/ports/psoc6/run_psoc6_tests.sh --test-suite pdm_pcm --board ${{ matrix.board }} --hil ${{ runner.name }}
```
Comment (jaenrig-ifx) [inline .github/workflows/ports_psoc6.yml:75]: Please remove commented code.

File: ports/psoc6/machine_pdm_pcm.c [hunk 2]
```diff
@@ -511,21 +463,7 @@ static void mp_machine_pdm_pcm_init_helper(machine_pdm_pcm_obj_t *self, mp_arg_v
     self->sample_rate = args[ARG_sample_rate].u_int;
     self->decimation_rate = args[ARG_decimation_rate].u_int;
 
-    // Set clock values
```
Comment (jaenrig-ifx) [inline ports/psoc6/machine_pdm_pcm.c:514]: I am missing a check where we raise an exception if the clock is not properly configured for the desired rate?

File: tests/ports/psoc6/run_psoc6_tests.sh [hunk 3]
```diff
@@ -341,6 +347,17 @@ run_ci_tests() {
       fi
     fi
     i2s_tests
+
+    if [ "${board}" == "CY8CPROTO-062-4343W" ] || [ "${board}" == "CY8CPROTO-063-BLE" ]; then
+      dev_test=${devs_b[0]}
```
Comment (jaenrig-ifx) [inline tests/ports/psoc6/run_psoc6_tests.sh:352]: https://github.com/Infineon/micropython/pull/195/files#diff-cafe99212312cbfc6caa307a421984d9049bb0145883517444e505039fc46aebR287-R307

Here we have to update the version to 0.6.0 (only for CY8CPROTO-062-4343W) to match the diagram version.
And in the {hil-name}-devs.yml file we have to also update that version.

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
All very nice 👯 
Some tests are now failing as I added the pdm connection, but I will get it out of the HIL and ensure it runs properly. 
The cabling is not the most reliable... and I might have broken proper contact between some of these jumper cables.

Meanwhile, only a few comments and changes to fully wrap the PR 🔧

File: tests/ports/psoc6/run_psoc6_tests.sh (hunk 3 above)
Comment (NikhitaR-IFX) [reply tests/ports/psoc6/run_psoc6_tests.sh:355]: Done :)

File: .github/workflows/ports_psoc6.yml (hunk 1 above)
Comment (NikhitaR-IFX) [reply .github/workflows/ports_psoc6.yml]: Done

File: ports/psoc6/machine_pdm_pcm.c (hunk 2 above)
Comment (NikhitaR-IFX) [reply ports/psoc6/machine_pdm_pcm.c:514]: Sorry it was a miss when I wanted to bring clock related changes. It's back there now!

**Review by IFX-Anusha — APPROVED** [review]

**Pull request merged**